### PR TITLE
#166816092 Setup Base application deployment to Heroku

### PR DESCRIPTION
**Pull request by fxola**
#### What does this PR do?
- Setup Base application deployment to Heroku

#### Description of Task to be completed?
- Create Procfile for heroku deployment
- Add scripts to enable deployment in package.json file
- Install babel packages to enable transpiling of es6 syntax
- Add .babelrc file to enable es6 transpilation
- Refactor folder structure into a src folder
- Update test script
- Update .gitignore file to include dist folder
- delivers[#166816092]

#### How should this be manually tested?
- Visit https://persephone-backend.herokuapp.com 

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- PT Story link - [#166816092](https://www.pivotaltracker.com/n/projects/2353116/stories/166816092)

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+});
+
+const PORT = process.env.PORT || 3000;
+
+// finally, let's start our server...
+app.listen(PORT, function() {
+  console.log(`Listening on port: ${PORT}`);
+});
+
+module.exports = { app };
```
Comment (nedemenang) [inline src/index.js]: what does the style guide say about exporting modules? Check and make necessary changes

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import bodyParser from 'body-parser';
+
+// Create global app object
+const app = express();
+
+// Normal express config defaults
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
+
+app.get('/', (req, res) => {
```
Comment (nedemenang) [inline src/index.js]: No shorthand. Use `request` and `response`

File: src/tests/index.test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* eslint-env mocha */
+import chai from 'chai';
+
+import chaiHttp from 'chai-http';
+
+const { app } = require('../index');
```
Comment (nedemenang) [inline src/tests/index.test.js]: Use `import` rather than `require`

File: src/tests/index.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const { expect } = chai;
+
+chai.use(chaiHttp);
+
+describe('First test', () => {
+  it('Should test to see if App is Up', () => {
+    chai
+      .request(app)
+      .get('/')
+      .end((err, res) => {
```
Comment (nedemenang) [inline src/tests/index.test.js]: No shorthand. Use `error` and `response`

**Review by nedemenang — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (fxola) [reply src/index.js]: Don't know how i didn't notice that, thank you

File: src/index.js (hunk 2 above)
Comment (fxola) [reply src/index.js]: Alright

**Review by nedemenang — APPROVED** [review]

**Pull request merged**